src/main/java/edu/wpi/grip/ui/preview/SocketPreviewView.java

```diff
@@ -40,22 +32,7 @@ public SocketPreviewView(EventBus eventBus, Socket<T> socket) {
         this.setCollapsible(false);
     }
 
-    public Socket<T> getSocket() {
+    public OutputSocket<T> getSocket() {
```

Comment:
Change this signature to protected and the name should probably be `getOutputSocket()`